src/components/scenes/CreateWalletSelectFiatScene.tsx

```diff
@@ -1,161 +1,225 @@
 import { SceneHeader } from '../themed/SceneHeader'
 import { SelectableRow } from '../themed/SelectableRow'
+import { WalletCreateItem } from '../themed/WalletList'
 
 type OwnProps = {
   navigation: NavigationProp<'createWalletSelectFiat'>
   route: RouteProp<'createWalletSelectFiat'>
 }
-type StateProps = {
-  supportedFiats: GuiFiatType[]
-}
-type Props = OwnProps & StateProps & ThemeProps
 
-type State = {
-  searchTerm: string
-  selectedFiat: string
-}
+type Props = OwnProps
 
-export class CreateWalletSelectFiatComponent extends React.Component<Props, State> {
-  constructor(props: Props) {
-    super(props)
-    this.state = {
-      searchTerm: '',
-      selectedFiat: ''
-    }
-  }
+const CreateWalletSelectFiatComponent = (props: Props) => {
+  const { navigation, route } = props
+  const { createWalletList } = route.params
 
-  isValidFiatType = () => {
-    const { selectedFiat } = this.state
-    const fiatTypeIndex = this.props.supportedFiats.findIndex(fiatType => fiatType.value === selectedFiat)
-    const isValid = fiatTypeIndex >= 0
-    return isValid
-  }
+  const dispatch = useDispatch()
+  const theme = useTheme()
+  const styles = getStyles(theme)
 
-  getFiatType = (fiatKey: string) => {
-    const fiatTypeIndex = this.props.supportedFiats.findIndex(fiatType => fiatType.value === fiatKey)
+  const account = useSelector(state => state.core.account)
+  const currencyWallets = useWatch(account, 'currencyWallets')
 
-    return this.props.supportedFiats[fiatTypeIndex]
-  }
+  const defaultFiat = useSelector(state => getDefaultFiat(state))
+  const [fiat, setFiat] = React.useState(getSupportedFiats(defaultFiat)[0])
+
+  const { newWalletItems, newTokenItems } = React.useMemo(() => splitCreateWalletItems(createWalletList), [createWalletList])
+
+  const defaultWalletNames = React.useMemo(
+    () =>
+      createWalletList.reduce<{ [key: string]: string }>((map, item) => {
+        map[item.key] = sprintf(s.strings.my_crypto_wallet_name, item.displayName)
+        return map
+      }, {}),
+    [createWalletList]
+  )
 
-  onNext = () => {
-    const { navigation, route } = this.props
-    const { cleanedPrivateKey, selectedWalletType } = route.params
-
-    if (this.isValidFiatType()) {
-      // check if account-based or not
-      const specialCurrencyInfo = getSpecialCurrencyInfo(selectedWalletType.walletType)
-      // check if eos-like
-      if (!specialCurrencyInfo.needsAccountNameSetup || cleanedPrivateKey) {
-        navigation.navigate('createWalletName', {
-          selectedWalletType: selectedWalletType,
-          selectedFiat: this.getFiatType(this.state.selectedFiat),
-          cleanedPrivateKey
-        })
-      } else {
-        navigation.navigate('createWalletAccountSetup', {
-          selectedWalletType: selectedWalletType,
-          selectedFiat: this.getFiatType(this.state.selectedFiat)
-        })
+  const [walletNames, setWalletNames] = React.useState(defaultWalletNames)
```

Comment:
For the `defaultWalletNames` thing, you can just put the function inside the `React.useState` and skip the whole `React.useMemo`. The function will run on first render, since that is a built-in `React.useState` feature.